src/pages/patientView/mutation/PatientViewMutationTable.tsx

```diff
@@ -100,6 +114,40 @@ export default class PatientViewMutationTable extends MutationTable<IPatientView
             resizable: true,
         };
 
+        // customization for ASCN-related columns
+        // Patient view differs from default/results view because multiple samples can appear in a row
+        // due to same variant in multiple samples
+        // This can lead to cases where there are multiple icons/tooltips in a single cell
+        // therefore patient view needs sampleManager to indicate which values match which samples
+        this._columns[MutationTableColumnType.CANCER_CELL_FRACTION] = {
+            name: "CCF",
+            tooltip:(<span>Cancer Cell Fraction</span>),
+            render: (d:Mutation[])=>PatientCancerCellFractionColumnFormatter.renderFunction(d, this.props.sampleManager),
+            sortBy:(d:Mutation[])=>d.map(m=>m.alleleSpecificCopyNumber.ccfMCopies),
+            download:(d:Mutation[])=>CancerCellFractionColumnFormatter.getCancerCellFractionDownload(d)
+        };
+
+        this._columns[MutationTableColumnType.CLONAL] = {
+            name: "Clonal",
+            tooltip: (<span>FACETS Clonal</span>),
+            render:(d:Mutation[])=>PatientClonalColumnFormatter.renderFunction(d, this.getSamples(), this.props.sampleManager),
+            sortBy:(d:Mutation[])=>d.map(m=>m.alleleSpecificCopyNumber.ccfMCopiesUpper),
+            download:(d:Mutation[])=>PatientClonalColumnFormatter.getClonalDownload(d)
+        };
+
+        this._columns[MutationTableColumnType.MUTANT_COPIES] = {
+            name: "Mutant Copies",
+            tooltip: (<span>FACETS Best Guess for Mutant Copies / Total Copies</span>),
+            render:(d:Mutation[])=>PatientMutantCopiesColumnFormatter.renderFunction(d, this.props.sampleIdToClinicalDataMap, this.getSamples(), this.props.sampleManager),
+            download:(d:Mutation[])=>MutantCopiesColumnFormatter.getMutantCopiesDownload(d, this.props.sampleIdToClinicalDataMap),
+            sortBy:(d:Mutation[])=>MutantCopiesColumnFormatter.getDisplayValueAsString(d, this.props.sampleIdToClinicalDataMap, this.getSamples())
+        };
+
+        this._columns[MutationTableColumnType.FACETS_COPY_NUM] = {
```

Comment:
I don't see this column in your ticket, could you add this screenshot as well?

Reply:
@zhx828 added! this column is Integer Copy #